ios/RCTTWVideoModule.m

```diff
@@ -176,26 +156,44 @@ - (void)addParticipantView:(TVIVideoView *)view sid:(NSString *)sid trackSid:(NS
 }
 
 - (void)startCameraCapture:(NSString *)cameraType {
-  if (self.camera == nil) {
-    return;
-  }
-  AVCaptureDevice *camera;
+    if (self.camera == nil) {
+        return;
+    }
+    AVCaptureDevice *camera;
     if ([cameraType isEqualToString:@"back"]) {
-    camera = [TVICameraSource captureDeviceForPosition:AVCaptureDevicePositionBack];
-  } else {
-    camera = [TVICameraSource captureDeviceForPosition:AVCaptureDevicePositionFront];
-  }
+        camera = [TVICameraSource captureDeviceForPosition:AVCaptureDevicePositionBack];
+    } else {
+        camera = [TVICameraSource captureDeviceForPosition:AVCaptureDevicePositionFront];
+    }
+    [self.camera startCaptureWithDevice:camera format: [self videoFormat:camera] completion:^(AVCaptureDevice *device,
+                                                                                              TVIVideoFormat *startFormat,
+                                                                                              NSError *error) {
+        if (!error) {
+            for (TVIVideoView *renderer in self.localVideoTrack.renderers) {
+                [self updateLocalViewMirroring:renderer];
+            }
+            [self sendEventCheckingListenerWithName:cameraDidStart body:nil];
+        }
+    }];
+}
 
-  [self.camera startCaptureWithDevice:camera completion:^(AVCaptureDevice *device,
-          TVIVideoFormat *startFormat,
-          NSError *error) {
-      if (!error) {
-          for (TVIVideoView *renderer in self.localVideoTrack.renderers) {
-            [self updateLocalViewMirroring:renderer];
-          }
-          [self sendEventCheckingListenerWithName:cameraDidStart body:nil];
-      }
-  }];
+-(TVIVideoFormat *)videoFormat: (AVCaptureDevice *) device {
+    TVIVideoFormat *selectedFormat = nil;
+    NSOrderedSet<TVIVideoFormat *> *formats = [TVICameraSource supportedFormatsForDevice:device];
+
+    for (TVIVideoFormat *format in formats) {
+        if (format.pixelFormat != TVIPixelFormatYUV420BiPlanarFullRange) {
+            continue;
+        }
+        selectedFormat = format;
+        CMVideoDimensions dimensions = format.dimensions;
+
+        if (dimensions.width >= videoDimensions.width && dimensions.height >= videoDimensions.height) {
```

Comment:
@andreychernikovich you also need to check that selectedFormat.frameRate is >= of the requested frame rate.